src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,232 @@
+  private static final String NETWORK_INTERFACE_CONFIG = "ONE_TO_ONE_NAT";
+  private static final String NETWORK_ACCESS_CONFIG = "External NAT";
+  private static final long OPERATION_TIMEOUT_MILLIS = 60 * 1000;
+  private static HttpTransport httpTransport;
+  private static final JsonFactory JSON_FACTORY = 
+    JacksonFactory.getDefaultInstance();
+  private static GoogleCredentials credential;
+  private  Compute compute;
+
+  /** 
+   * Creates an environment, calls each operation and tracks to see if they 
+   * are executed.
+   * @param {String} operation - the name of the method that should be called.
+   * @param {String} instanceName - the name of the instance being operated on.
+   * @param {Optional<String>} password - the password needed to access this 
+   *    instance. Not all operations requires a password.
+   */
+  public void helper(String operation, String instanceName,
+    Optional<String> password) {
+      try {
+        httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        credential = GoogleCredentials.getApplicationDefault();
+        if (credential.createScopedRequired()) {
+          List<String> scopes = new ArrayList<>();
+          scopes.add(ComputeScopes.DEVSTORAGE_FULL_CONTROL);
+          scopes.add(ComputeScopes.COMPUTE);
+          credential = credential.createScoped(scopes);
+        }
+        HttpRequestInitializer requestInitializer = 
+          new HttpCredentialsAdapter(credential);
+        // Create Compute Engine object.
+        compute = new Compute.Builder
+          (httpTransport, JSON_FACTORY, requestInitializer)
+          .setApplicationName(APPLICATION_NAME)
+          .build();
+        Operation op = null;
+        if (operation == "createInstance") {
+          op = createInstance(instanceName, password.get());
+        }
+        else if (operation == "deleteInstance") {
+          op = deleteInstance(instanceName);
+        }
+        else if (operation == "stopInstance") {
+          op = stopInstance(instanceName);
+        }
+        else if (operation == "restartInstance") {
+          op = restartInstance(instanceName, password.get());
+        }
+        else {
+          System.out.println("Operation not found!");
+        }
+        // Call Compute Engine API operation and poll for operation completion 
+        // status
+        System.out.println("Waiting for operation completion...");
+        Operation.Error error = blockUntilComplete
+          (compute, op, OPERATION_TIMEOUT_MILLIS);
+        if (error == null) {
+          System.out.println("Success!");
+        } else {
+          System.out.println(error.toPrettyString());
+        }
+      } catch (IOException e) {
+          System.out.println("Got error!");
+          System.err.println(e.getMessage());
+      } catch (Throwable t) {
+          t.printStackTrace();
+        }
+    }
+    
+  // [START create_instances]
+  public Operation createInstance(String instanceName, String password)
```

Comment:
if the password is meant for the vncserver, i am not sure what this customerEncryptionKey business is, you can use Metadata like you did for the startup script to set the password there